src/components/Table/Defi/Protocols/ProProtocols.tsx

```diff
@@ -0,0 +1,545 @@
+	getFilteredRowModel
+} from '@tanstack/react-table'
+import VirtualTable from '~/components/Table/Table'
+import {
+	columnOrders,
+	columnSizes,
+	protocolAddlColumns,
+	protocolsColumns,
+	recentlyListedProtocolsColumns,
+	topGainersAndLosersColumns,
+	protocolsByTokenColumns,
+	airdropsColumns,
+	protocolsByChainColumns
+} from './columns'
+import useWindowSize from '~/hooks/useWindowSize'
+import { IProtocolRow } from './types'
+import { useRouter } from 'next/router'
+import { SearchIcon, TableFiltersWithInput } from '../../shared'
+import styled from 'styled-components'
+import { TVLRange } from '~/components/Filters'
+import { ColumnFilters2 } from '~/components/Filters/common/ColumnFilters'
+import RowFilter from '~/components/Filters/common/RowFilter'
+
+const columnSizesKeys = Object.keys(columnSizes)
+	.map((x) => Number(x))
+	.sort((a, b) => Number(b) - Number(a))
+
+export function ProtocolsTable({
+	data,
+	addlColumns,
+	removeColumns
+}: {
+	data: Array<IProtocolRow>
+	addlColumns?: Array<string>
+	removeColumns?: Array<string>
+}) {
+	const [sorting, setSorting] = React.useState<SortingState>([{ desc: true, id: 'tvl' }])
+	const [columnOrder, setColumnOrder] = React.useState<ColumnOrderState>([])
+	const [columnSizing, setColumnSizing] = React.useState<ColumnSizingState>({})
+	const [expanded, setExpanded] = React.useState<ExpandedState>({})
+	const windowSize = useWindowSize()
+
+	const columnsData = React.useMemo(
+		() =>
+			addlColumns || removeColumns
+				? [
+						...protocolsColumns.filter((c) => !(removeColumns ?? []).includes((c as any).accessorKey)),
+						...(addlColumns ?? []).map((x) => protocolAddlColumns[x])
+				  ]
+				: protocolsColumns,
+		[addlColumns, removeColumns]
+	)
+
+	const instance = useReactTable({
+		data,
+		columns: columnsData,
+		state: {
+			sorting,
+			expanded,
+			columnOrder,
+			columnSizing
+		},
+		onExpandedChange: setExpanded,
+		getSubRows: (row: IProtocolRow) => row.subRows,
+		onSortingChange: setSorting,
+		onColumnOrderChange: setColumnOrder,
+		onColumnSizingChange: setColumnSizing,
+		getCoreRowModel: getCoreRowModel(),
+		getSortedRowModel: getSortedRowModel(),
+		getExpandedRowModel: getExpandedRowModel()
+	})
+
+	React.useEffect(() => {
+		const defaultOrder = instance.getAllLeafColumns().map((d) => d.id)
+
+		const order = windowSize.width
+			? columnOrders.find(([size]) => windowSize.width > size)?.[1] ?? defaultOrder
+			: defaultOrder
+
+		const cSize = windowSize.width
+			? columnSizesKeys.find((size) => windowSize.width > Number(size))
+			: columnSizesKeys[0]
+
+		instance.setColumnSizing(columnSizes[cSize])
+
+		instance.setColumnOrder(order)
+	}, [windowSize, instance])
+
+	return <VirtualTable instance={instance} />
+}
+
+enum TABLE_CATEGORIES {
+	FEES = 'Fees',
+	REVENUE = 'Revenue',
+	VOLUME = 'Volume',
+	TVL = 'TVL'
+}
+
+enum TABLE_PERIODS {
+	ONE_DAY = '1d',
+	SEVEN_DAYS = '7d',
+	ONE_MONTH = '1m'
+}
+
+const protocolsByChainTableColumns = [
```

Comment:
this is a exact copy of the variable that is in src/components/Table/Defi/Protocols/index.tsx, but its a copy of an old version where the bug around ps/pf ad not been fixed yet.

It highlights how having duplicated code can lead to more bugs, because bugs might be fixed on one instance but not the other

Please review the code for other instances of duplication and try to squash them into a single instance, if you need to support new behaviour you can just extend previous code with more functionality